### ISSUE TITLE
Date in version causes orphaned CSV in CatalogSource

### ISSUE THREAD

**Issue by jewzaam** [openshift/dedicated-admin-operator#28]
If we have date in the version a build of a catalog source in osd-operators-registry will have a version of the operator image that does NOT exist in a repo somewhere.  Additionally, the catalog will include a new version that has no functional changes.  The only change is the date therefore only the version.

1. build & push operator today, version is 0.1.73-2019-04-11-4086f670
2. build osd-operators-registry today, references operator version 0.1.73-2019-04-11-4086f670
3. build osd-operators-registry tomorrow, references operator version 0.1.73-2019-04-**12**-4086f670

After step 3 the registry no longer points to a valid version.

I propose we remove date from the version.

@lisa

**Comment by lisa** [comment]
There's a lot of variance in the version scheme, with the git commit count changing, the git sha changing and the date changing. Perhaps we ought to consider changing the scheme? If we tag releases we no longer need the git sha, nor the commit count, nor even the date.

**Comment by jewzaam** [comment]
That's reasonable.  For any given commit only the date is the thing that could change, so that's what I'm focused on.  I think we should solve that quickly before it bites us more.  Could we open discussion in another issue on the version standard overall?

**Comment by lisa** [comment]
@jewzaam yeah, definitely worth looking at. I spent some time thinking about this last night.

**Comment by jewzaam** [comment]
Fixed with https://github.com/openshift/dedicated-admin-operator/pull/29
